### ISSUE TITLE
Store response for newly created resources

### ISSUE THREAD

**Issue by mjallday** [deontologician/restnavigator#42]
IF I call `POST /users` I cannot get the result of the initial call if the `Location` header is set. 

This is difficult for me because this call to `/users` returns a set of credentials for operating on the API and is subsequently needed for authentication. 

What's the recommended way to work around this? I see that in the source code the [response is discarded](https://github.com/deontologician/restnavigator/blob/next/restnavigator/halnav.py#L559-L560) if the Location header is set regardless of if there is a body or not.

According to this [wikipedia article about the Location header](https://en.wikipedia.org/wiki/HTTP_location) should always be sent on 201 responses.

> To provide information about the location of a newly created resource. In this circumstance, the Location header should be sent with an HTTP status code of 201 or 202.

**Comment by deontologician** [comment]
Hey @mjallday currently it just does a redirection automatically. The idea being you aren't really interested in the 201 response, just in the location of the newly created resource. Of course, in real life, you often can't control the api itself, so this behavior is annoying (sorry).

I had a start on a similar issue in #29 but I haven't finished it.

**Comment by mjallday** [comment]
Right. What's the thinking with [this line](https://github.com/deontologician/restnavigator/blob/next/restnavigator/halnav.py#L550)? I'm not too familiar with HTTP APIs that don't return a body on a 201. Seems like 202 would be more accurate in this case?

**Comment by deontologician** [comment]
The idea is that a resource has been created, and we want to return you a
navigator pointing to the final resource, instead of to the response to the
creation request. According to the http definition, the body the 201 should
just be links or descriptive properties about the created resource, with
the uri in the location header.

Again, that's the spec, in reality if your api retuns stuff in the 201 body
that you need, rest navigator should let you get at it

**Comment by Skizu** [comment]
Following on from this, assuming that you're already using an authenticated navigator this should not be an issue. Using the next branch you would simply be able to do the following.

```pythonN = restnavigator.Navigator.hal(api, auth=(user, password))`

**Comment by Skizu** [comment]
Following on from this, assuming that you're already using an authenticated navigator this should not be an issue. Using the next branch you would simply be able to do the following.

``` python
N = restnavigator.Navigator.hal(api, auth=(user, password))
N_new_user = N['users'].create(payload)
```

Then proceed to use the N_new_user navigator as you please, as although the create will not bind the result, it will fetch the data upon use.
